### PR TITLE
Enriched scope and path

### PR DESCRIPTION
**Pull request by evstratbg**
Hi!

I added headers to the `scope` for `get_router_path` as I have my custom Router which matches against headers, so-called "header-based versioning", as well as deleted `Mount` isinstance check, because startlette allows Routers include Routers as paths
Would be cool if you accept this PR and release a patch version of the package

Looking forward to your comments

### CODE REVIEW

File: starlette_exporter/middleware.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ def get_matching_route_path(
             # be the complete path (it may represent the path to the
             # mounted router). If this is a mounted route, descend into it to
             # get the complete path.
-            if isinstance(route, Mount) and route.routes:
+            if hasattr(route, "routes"):
```
Comment (evstratbg) [inline starlette_exporter/middleware.py:43]: another option here is to have
```suggestion
if isinstance(route, BaseRoute) and route.routes:
```

**Comment by stephenhillier** [comment]
Hi @evstratbg , I'm curious why the StaticFile test fails when switching the isinstance check from Mount to BasePath. Any idea?

**Comment by evstratbg** [comment]
> Hi @evstratbg , I'm curious why the StaticFile test fails when switching the isinstance check from Mount to BasePath. Any idea?

Hi! I cant reproduce your issue. Are you sure you meant `BasePath` and not `BaseRoute`? with `BaseRoute` everything works as shown in CI and I dont see any challenges as Mount inherits BaseRoute, eventually

Can you share the details of your setup?

**Comment by stephenhillier** [comment]
Yes, I meant BaseRoute. I'm not having an issue, I just noticed that StaticFiles handler setup changed at the same time as moving it to testconf.py. When I change it back, the test fails. 

Also, thank you for contributing! Sorry it's taken some time to get to this, things have been extra busy.

**Comment by evstratbg** [comment]
@stephenhillier it hasn't changed, I just moved it to a `conftest.py`, because it's not a test, but the fixture, that is needed for tests.

What else can I do to get this MR merged?

**Comment by stephenhillier** [comment]
>@stephenhillier it hasn't changed, I just moved it to a conftest.py

You made a change to the StaticFile test setup at the same time that you moved the files around. I would like to help support your use case but the tests need to pass without modifications (they currently don't).

Happy to take another look if you can get the tests to pass as-is, or if you can share more about the header based routing I can try to help find a way to support it.

**Comment by evstratbg** [comment]
@stephenhillier I don't quite understand what you are talking about, because the latest checks have been [passed](https://github.com/stephenhillier/starlette_exporter/pull/76/checks?sha=6ab98a971a865f87b79337ca2295ab590b7ebef4), all green

I rollbacked my changes to the `tests` folder to keep is as-is

**Comment by stephenhillier** [comment]
You are correct, they are passing. I will try to do a release as soon as I can this week. Thank you for contributing this.

Just for my own knowledge, do you have any examples of the header based versioning? Is this a library on GitHub or custom middleware?

**Comment by evstratbg** [comment]
@stephenhillier thanks!
I am planning to publish it this week. I can ping you here with a link if you dont mind

**Comment by evstratbg** [comment]
I started preparing the repo, you can find it [here](https://github.com/team-monite/verselect)
Planning to finish this week

**Comment by evstratbg** [comment]
released first version

**Comment by evstratbg** [comment]
> I started preparing the repo, you can find it [here](https://github.com/team-monite/verselect) Planning to finish this week

released first version

**Pull request merged**